erpnext/stock/doctype/serial_no/serial_no.py

```diff
@@ -402,10 +402,16 @@ def update_serial_nos(sle, item_det):
 def get_auto_serial_nos(serial_no_series, qty):
 	serial_nos = []
 	for i in range(cint(qty)):
-		serial_nos.append(make_autoname(serial_no_series, "Serial No"))
+		serial_nos.append(create_sno(serial_no_series))
 
 	return "\n".join(serial_nos)
 
+def create_sno(serial_no_series):
+	sno = make_autoname(serial_no_series, "Serial No")
+	if frappe.db.exists("Serial No", sno):
+		sno = create_sno(serial_no_series)
+	return sno
```

Comment:
overall fix is okay; some suggestions on naming:

function should be probably be called "get_new_serial_number", along with argument "series" it makes it pretty clear as to what it does. 

Also `sno` is bad variable name; `sr_no` maybe?

---

up to you: Add a test case for this behaviour. (just encode the steps you described in a unit test 😄)